### PR TITLE
[10.0] base_rest: Add swagger frontend and generate OpenAPI specs for sevices 

### PR DESCRIPTION
**Pull request by lmignon**
based  on #3 
moved from https://github.com/akretion/odoo-shopinvader/pull/144
* [X] Add a swagger interface to test the services
* [X] Generate openapi definition for each services
image
image

* [X] Backward compatibility? (Response schema is required)
* [X] Missing tests
* [X] Code cleaning
* [X] Binary format for request/response  (ie post or retrieve a file)

### CODE REVIEW

File: base_rest/components/service.py [hunk 1]
```diff
@@ -151,8 +150,7 @@ def _secure_output(self, method, response):
         v = Validator(schema, purge_unknown=True)
         if v.validate(response):
             return v.document
-        _logger.error("Invalid response %s", v.errors)
-        raise UserError(_('Invalid Response'))
+        raise UserError(_('Invalid Response %s') % v.errors)
```
Comment (sbidoul) [inline base_rest/components/service.py]: IMO output validation error is a system error, not a user error

Comment (sbidoul) [reply base_rest/components/service.py]: because it's the developer's fault, not the user's

Comment (lmignon) [reply base_rest/components/service.py]: it makes sense. I'll change

**Comment by sbidoul** [comment]
I think most of the TODO's in the PR description are done right?

**Review by sbidoul — APPROVED** [review]
This has been used successfully in several projects.

Thanks for this great framework.

**Comment by lmignon** [comment]
@sbidoul Yes they are done.

**Review by sebastienbeau — APPROVED** [review]
Thansk for this awsome work, used in production

**Comment by pedrobaeza** [comment]
This is not in v11 yet, right?

**Comment by lmignon** [comment]
@pedrobaeza v10 and v11 provides the same features and share the same code. IOW Both versions are aligned

**Comment by pedrobaeza** [comment]
Thanks for the confirmation!

**Pull request merged**